cmd/gossamer/gossamer_test.go

```diff
@@ -0,0 +1,154 @@
+package main_test
+
+import (
+	"encoding/json"
+	"fmt"
+	"go/build"
+	"os"
+	"os/exec"
+	"path"
+	"reflect"
+	"strings"
+	"testing"
+	"time"
+
+	cfg "github.com/ChainSafe/gossamer/config"
+	"github.com/ChainSafe/gossamer/p2p"
+	"github.com/ChainSafe/gossamer/polkadb"
+	"github.com/inconshreveable/log15"
+	"github.com/rendon/testcli"
+)
+
+var binaryname = "gossamer-test"
+
+const configTest = "config-test.toml"
+const timeFormat = "2006-01-02T15:04:05-0700"
+
+func setup() {
+	gopath := os.Getenv("GOPATH")
+	if gopath == "" {
+		gopath = build.Default.GOPATH
+	}
+	r := exec.Command("go", "build", "-o", gopath+"/bin/gossamer-test")
+	err := r.Run()
+	if err != nil {
+		log15.Crit("could not execute binary", "executable", binaryname, "err", err)
+		os.Exit(1)
+	}
+	run := exec.Command(`gossamer-test`)
+	err = run.Run()
+	if err != nil {
+		log15.Crit("could not execute binary", "executable", binaryname, "err", err)
+		os.Exit(1)
+	}
+	fp, err := os.Create(configTest)
```

Comment:
I think it would be safer to make use of explicitly temporary files (see https://golangcode.com/creating-temp-files/).